connection/src/test/java/com/netcracker/ConnectionTest.java

```diff
@@ -0,0 +1,112 @@
+    private static final Message messageToSend = new Message(TEXT, "MessageToSend");
+    @Rule
+    public final ExpectedException exception = ExpectedException.none();
+
+    @Before
+    public void setUpConnectionObjectWithByteStreams() throws IOException {
+        ByteArrayOutputStream outerOutputStream = new ByteArrayOutputStream();
+        try (ObjectOutputStream outerOOS = new ObjectOutputStream(outerOutputStream)) {
+            outerOOS.writeObject(messageToReceive);
+        }
+        ByteArrayInputStream innerInputStream = new ByteArrayInputStream(outerOutputStream.toByteArray());
+
+        ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+        try (ObjectOutputStream oos = new ObjectOutputStream(buffer)) {
+            oos.writeObject(new Message(TEXT, "InOrderNotToGetEOFException"));
+        }
+        byte[] rawData = buffer.toByteArray();
+        originalNumberOfBytes = rawData.length;
+
+        innerOutputStream = new ByteArrayOutputStream();
+        innerOutputStream.write(rawData);
+
+        Socket socket = mock(Socket.class);
+        when(socket.getOutputStream()).thenReturn(innerOutputStream);
+        when(socket.getInputStream()).thenReturn(innerInputStream);
+        when(socket.getRemoteSocketAddress()).thenReturn(address);
+
+        connection = new Connection(socket);
+    }
+
+    public void setUpConnectionWithFileStream() throws IOException {
+        FileOutputStream buffer = new FileOutputStream("src\\test\\" +
+                "java\\com\\netcracker\\TestResource");
+
+        try (ObjectOutputStream oos = new ObjectOutputStream(buffer)) {
+            oos.writeObject(messageToSend);
+        }
+
+        FileInputStream innerInputStream = new FileInputStream("src\\test\\" +
+                "java\\com\\netcracker\\TestResource");
+
+        FileOutputStream innerOutputStream = new FileOutputStream("src\\test\\" +
+                "java\\com\\netcracker\\TestResource");
+
+        Socket socket = mock(Socket.class);
+        when(socket.getOutputStream()).thenReturn(innerOutputStream);
+        when(socket.getInputStream()).thenReturn(innerInputStream);
+        when(socket.getRemoteSocketAddress()).thenReturn(address);
+
+        connection = new Connection(socket);
+    }
+
+    @Test
+    public void shouldSendMessageWhenAskingForIt() throws IOException, ClassNotFoundException {
+        connection.send(messageToSend);
+
+        assertTrue(innerOutputStream.toByteArray().length > originalNumberOfBytes);
+    }
+
+    @Test
+    public void shouldReceiveMessageWhenAskingForIt() throws IOException, ClassNotFoundException {
+        assertEquals(messageToReceive.getData(), connection.receive().getData());
+    }
+
+    @Test
+    public void shouldReturnRemoteSocketAddressWhenAskingForIt() {
+        assertEquals(address, connection.getRemoteSocketAddress());
+    }
+
+    @Test
+    public void canNotGetMessagesWhenConnectionClosed() throws ClassNotFoundException, IOException {
```

Comment:
&#x1F34E; **Issue**
In tests you should just throw Exception instead of several specific exceptions to make things more readable.
You don't care whether your test has thrown IOException or some other exception.